// ng-uikit-pro-standard/esm5/lib/pro/sticky-content/sticky-content.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { CommonModule } from '@angular/common';
import { MdbStickyDirective } from './sticky-content.directive';
import * as ɵngcc0 from '@angular/core';
export { MdbStickyDirective };
var StickyContentModule = /** @class */ (function () {
    function StickyContentModule() {
    }
StickyContentModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: StickyContentModule });
StickyContentModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function StickyContentModule_Factory(t) { return new (t || StickyContentModule)(); }, imports: [[CommonModule, FormsModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(StickyContentModule, { declarations: function () { return [MdbStickyDirective]; }, imports: function () { return [CommonModule, FormsModule]; }, exports: function () { return [MdbStickyDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(StickyContentModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, FormsModule],
                declarations: [MdbStickyDirective],
                exports: [MdbStickyDirective]
            }]
    }], function () { return []; }, null); })();
    return StickyContentModule;
}());
export { StickyContentModule };
