// ng-uikit-pro-standard/esm5/lib/pro/accordion/index.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SqueezeBoxComponent } from './components/squeezebox';
import { SBItemComponent } from './components/sb-item';
import { SBItemHeadComponent } from './components/sb-item.head';
import { SBItemBodyComponent } from './components/sb-item.body';
/** @type {?} */
import * as ɵngcc0 from '@angular/core';
export var SQUEEZEBOX_COMPONENTS = [SqueezeBoxComponent, SBItemComponent, SBItemHeadComponent, SBItemBodyComponent];
export { SBItemComponent } from './components/sb-item';
export { SBItemHeadComponent } from './components/sb-item.head';
export { SBItemBodyComponent } from './components/sb-item.body';
export { SqueezeBoxComponent } from './components/squeezebox';
var AccordionModule = /** @class */ (function () {
    function AccordionModule() {
    }
AccordionModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: AccordionModule });
AccordionModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function AccordionModule_Factory(t) { return new (t || AccordionModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(AccordionModule, { declarations: function () { return [SqueezeBoxComponent, SBItemComponent, SBItemHeadComponent, SBItemBodyComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [SqueezeBoxComponent, SBItemComponent, SBItemHeadComponent, SBItemBodyComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AccordionModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                declarations: [SQUEEZEBOX_COMPONENTS],
                exports: [SQUEEZEBOX_COMPONENTS]
            }]
    }], function () { return []; }, null); })();
    return AccordionModule;
}());
export { AccordionModule };
