// ng-uikit-pro-standard/esm5/lib/pro/accordion/mdb-accordion.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
var MdbAccordionService = /** @class */ (function () {
    function MdbAccordionService() {
        this._items = [];
        this._multiple = false;
    }
    /**
     * @param {?} item
     * @return {?}
     */
    MdbAccordionService.prototype.addItem = /**
     * @param {?} item
     * @return {?}
     */
    function (item) {
        this._items.push(item);
    };
    /**
     * @param {?} items
     * @return {?}
     */
    MdbAccordionService.prototype.updateItemsArray = /**
     * @param {?} items
     * @return {?}
     */
    function (items) {
        this._items = tslib_1.__spread(items);
    };
    /**
     * @param {?} value
     * @return {?}
     */
    MdbAccordionService.prototype.updateMultipleState = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        this._multiple = value;
    };
    /**
     * @param {?} item
     * @return {?}
     */
    MdbAccordionService.prototype.didItemToggled = /**
     * @param {?} item
     * @return {?}
     */
    function (item) {
        // on not multiple, it will collpase the rest of items
        if (!this._multiple) {
            this._items.forEach((/**
             * @param {?} el
             * @return {?}
             */
            function (el) {
                if (el !== item) {
                    el.applyToggle(true);
                }
                if (el === item) {
                    /** @type {?} */
                    var collapsed_1 = el.collapsed ? true : false;
                    setTimeout((/**
                     * @return {?}
                     */
                    function () {
                        el.applyToggle(collapsed_1);
                    }), 0);
                }
            }));
        }
    };
MdbAccordionService.ɵfac = function MdbAccordionService_Factory(t) { return new (t || MdbAccordionService)(); };
MdbAccordionService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: MdbAccordionService, factory: function (t) { return MdbAccordionService.ɵfac(t); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MdbAccordionService, [{
        type: Injectable
    }], function () { return []; }, null); })();
    return MdbAccordionService;
}());
export { MdbAccordionService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    MdbAccordionService.prototype._items;
    /**
     * @type {?}
     * @private
     */
    MdbAccordionService.prototype._multiple;
}
