// ng-uikit-pro-standard/esm5/lib/free/charts/chart.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { BaseChartDirective } from './chart.directive';
import * as ɵngcc0 from '@angular/core';
var ChartsModule = /** @class */ (function () {
    function ChartsModule() {
    }
ChartsModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ChartsModule });
ChartsModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ChartsModule_Factory(t) { return new (t || ChartsModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ChartsModule, { declarations: function () { return [BaseChartDirective]; }, exports: function () { return [BaseChartDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ChartsModule, [{
        type: NgModule,
        args: [{
                declarations: [BaseChartDirective],
                exports: [BaseChartDirective]
            }]
    }], function () { return []; }, null); })();
    return ChartsModule;
}());
export { ChartsModule };
