// swiper/angular/esm2015/angular/src/swiper.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SwiperComponent } from './swiper.component';
import { SwiperSlideDirective } from './swiper-slide.directive';
import * as ɵngcc0 from '@angular/core';
export class SwiperModule {
}
SwiperModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: SwiperModule });
SwiperModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function SwiperModule_Factory(t) { return new (t || SwiperModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SwiperModule, { declarations: function () { return [SwiperComponent, SwiperSlideDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [SwiperComponent, SwiperSlideDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SwiperModule, [{
        type: NgModule,
        args: [{
                declarations: [SwiperComponent, SwiperSlideDirective],
                exports: [SwiperComponent, SwiperSlideDirective],
                imports: [CommonModule]
            }]
    }], null, null); })();
