// swiper/angular/esm2015/angular/src/swiper-slide.directive.js
import { Directive, Input, TemplateRef } from '@angular/core';
import { coerceBooleanProperty } from './utils/utils';
import * as ɵngcc0 from '@angular/core';
export class SwiperSlideDirective {
    constructor(template) {
        this.template = template;
        this.class = '';
        this.slideData = {
            isActive: false,
            isPrev: false,
            isNext: false,
            isVisible: false,
            isDuplicate: false,
        };
    }
    set zoom(val) {
        this._zoom = coerceBooleanProperty(val);
    }
    get zoom() {
        return this._zoom;
    }
    get classNames() {
        return this._classNames;
    }
    set classNames(val) {
        if (this._classNames === val) {
            return;
        }
        this._classNames = val;
        this.slideData = {
            isActive: this._hasClass(['swiper-slide-active', 'swiper-slide-duplicate-active']),
            isVisible: this._hasClass(['swiper-slide-visible']),
            isDuplicate: this._hasClass(['swiper-slide-duplicate']),
            isPrev: this._hasClass(['swiper-slide-prev', 'swiper-slide-duplicate-prev']),
            isNext: this._hasClass(['swiper-slide-next', 'swiper-slide-duplicate-next']),
        };
    }
    _hasClass(classNames) {
        return classNames.some((className) => this._classNames.indexOf(className) >= 0);
    }
}
SwiperSlideDirective.ɵfac = function SwiperSlideDirective_Factory(t) { return new (t || SwiperSlideDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.TemplateRef)); };
SwiperSlideDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: SwiperSlideDirective, selectors: [["ng-template", "swiperSlide", ""]], inputs: { class: "class", zoom: "zoom", virtualIndex: "virtualIndex" } });
SwiperSlideDirective.ctorParameters = () => [
    { type: TemplateRef }
];
SwiperSlideDirective.propDecorators = {
    virtualIndex: [{ type: Input }],
    class: [{ type: Input }],
    zoom: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SwiperSlideDirective, [{
        type: Directive,
        args: [{
                selector: 'ng-template[swiperSlide]'
            }]
    }], function () { return [{ type: ɵngcc0.TemplateRef }]; }, { class: [{
            type: Input
        }], zoom: [{
            type: Input
        }], virtualIndex: [{
            type: Input
        }] }); })();
