// @nicky-lenaers/ngx-scroll-to/esm5/lib/scroll-to.module.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { ScrollToDirective } from './scroll-to.directive';
import { ScrollToService } from './scroll-to.service';
/** Scroll To Module */
import * as ɵngcc0 from '@angular/core';
var ScrollToModule = /** @class */ (function () {
    function ScrollToModule() {
    }
    ScrollToModule_1 = ScrollToModule;
    /**
     * Guaranteed singletons for provided Services across App.
     *
     * @return          An Angular Module with Providers
     */
    ScrollToModule.forRoot = function () {
        return {
            ngModule: ScrollToModule_1,
            providers: [
                ScrollToService
            ]
        };
    };
    var ScrollToModule_1;
ScrollToModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ScrollToModule });
ScrollToModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ScrollToModule_Factory(t) { return new (t || ScrollToModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ScrollToModule, { declarations: function () { return [ScrollToDirective]; }, exports: function () { return [ScrollToDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ScrollToModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    ScrollToDirective
                ],
                exports: [
                    ScrollToDirective
                ]
            }]
    }], function () { return []; }, null); })();
    return ScrollToModule;
}());
export { ScrollToModule };
