// ng-uikit-pro-standard/esm2015/lib/pro/accordion/mdb-accordion.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class MdbAccordionService {
    constructor() {
        this._items = [];
        this._multiple = false;
    }
    /**
     * @param {?} item
     * @return {?}
     */
    addItem(item) {
        this._items.push(item);
    }
    /**
     * @param {?} items
     * @return {?}
     */
    updateItemsArray(items) {
        this._items = [...items];
    }
    /**
     * @param {?} value
     * @return {?}
     */
    updateMultipleState(value) {
        this._multiple = value;
    }
    /**
     * @param {?} item
     * @return {?}
     */
    didItemToggled(item) {
        // on not multiple, it will collpase the rest of items
        if (!this._multiple) {
            this._items.forEach((/**
             * @param {?} el
             * @return {?}
             */
            (el) => {
                if (el !== item) {
                    el.applyToggle(true);
                }
                if (el === item) {
                    /** @type {?} */
                    const collapsed = el.collapsed ? true : false;
                    setTimeout((/**
                     * @return {?}
                     */
                    () => {
                        el.applyToggle(collapsed);
                    }), 0);
                }
            }));
        }
    }
}
MdbAccordionService.ɵfac = function MdbAccordionService_Factory(t) { return new (t || MdbAccordionService)(); };
MdbAccordionService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: MdbAccordionService, factory: MdbAccordionService.ɵfac });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MdbAccordionService, [{
        type: Injectable
    }], function () { return []; }, null); })();
if (false) {
    /**
     * @type {?}
     * @private
     */
    MdbAccordionService.prototype._items;
    /**
     * @type {?}
     * @private
     */
    MdbAccordionService.prototype._multiple;
}
