// angular-count-to/bundles/angular-count-to.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(require('@angular/core'),exports, require('@angular/core')) :
	typeof define === 'function' && define.amd ? define(['@angular/core','exports', '@angular/core'], factory) :
	(factory(global.ng.core,(global.ng = global.ng || {}, global.ng.angularCountTo = global.ng.angularCountTo || {}),global.ng.core));
}(this, (function (ɵngcc0,exports,_angular_core) { 'use strict';

var CountToDirective = (function () {
    /**
     * @param {?} el
     */
    function CountToDirective(el) {
        this.el = el;
        this.from = 0;
        this.duration = 4;
        this.e = this.el.nativeElement;
        this.refreshInterval = 30;
        this.step = 0;
    }
    /**
     * @return {?}
     */
    CountToDirective.prototype.ngOnInit = function () {
    };
    /**
     * @return {?}
     */
    CountToDirective.prototype.ngOnChanges = function () {
        if (this.CountTo) {
            this.start();
        }
    };
    /**
     * @return {?}
     */
    CountToDirective.prototype.calculate = function () {
        this.duration = this.duration * 1000;
        this.steps = Math.ceil(this.duration / this.refreshInterval);
        this.increment = ((this.CountTo - this.from) / this.steps);
        this.num = this.from;
    };
    /**
     * @return {?}
     */
    CountToDirective.prototype.tick = function () {
        var _this = this;
        setTimeout(function () {
            _this.num += _this.increment;
            _this.step++;
            if (_this.step >= _this.steps) {
                _this.num = _this.CountTo;
                _this.e.textContent = _this.CountTo;
            }
            else {
                _this.e.textContent = Math.round(_this.num);
                _this.tick();
            }
        }, this.refreshInterval);
    };
    /**
     * @return {?}
     */
    CountToDirective.prototype.start = function () {
        this.calculate();
        this.tick();
    };
CountToDirective.ɵfac = function CountToDirective_Factory(t) { return new (t || CountToDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
CountToDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: CountToDirective, selectors: [["", "CountTo", ""]], inputs: { from: "from", duration: "duration", CountTo: "CountTo" }, features: [ɵngcc0.ɵɵNgOnChangesFeature] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CountToDirective, [{
        type: _angular_core.Directive,
        args: [{
                selector: '[CountTo]'
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }]; }, { from: [{
            type: _angular_core.Input
        }], duration: [{
            type: _angular_core.Input
        }], CountTo: [{
            type: _angular_core.Input
        }] }); })();
    return CountToDirective;
}());
/**
 * @nocollapse
 */
CountToDirective.ctorParameters = function () { return [
    { type: _angular_core.ElementRef, },
]; };
CountToDirective.propDecorators = {
    'CountTo': [{ type: _angular_core.Input },],
    'from': [{ type: _angular_core.Input },],
    'duration': [{ type: _angular_core.Input },],
};
var CountToModule = (function () {
    function CountToModule() {
    }
    /**
     * Use in AppModule
     * @return {?}
     */
    CountToModule.forRoot = function () {
        return {
            ngModule: CountToModule,
            providers: []
        };
    };
    /**
     * Use in features modules with lazy loading
     * @return {?}
     */
    CountToModule.forChild = function () {
        return {
            ngModule: CountToModule,
            providers: []
        };
    };
CountToModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: CountToModule });
CountToModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function CountToModule_Factory(t) { return new (t || CountToModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(CountToModule, { declarations: [CountToDirective], exports: [CountToDirective] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CountToModule, [{
        type: _angular_core.NgModule,
        args: [{
                declarations: [
                    CountToDirective
                    // Pipes.
                    // Directives.
                ],
                exports: [
                    CountToDirective
                    // Pipes.
                    // Directives.
                ]
            }]
    }], function () { return []; }, null); })();
    return CountToModule;
}());
/**
 * @nocollapse
 */
CountToModule.ctorParameters = function () { return []; };

exports.CountToDirective = CountToDirective;
exports.CountToModule = CountToModule;

Object.defineProperty(exports, '__esModule', { value: true });

})));
