// ngx-youtube-player/esm2015/lib/ngx-youtube-player.module.js
import * as ɵngcc0 from '@angular/core';
var NgxYoutubePlayerModule_1;
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { YoutubePlayerComponent } from './ngx-youtube-player.component';
import { YoutubePlayerService } from './ngx-youtube-player.service';
let NgxYoutubePlayerModule = NgxYoutubePlayerModule_1 = class NgxYoutubePlayerModule {
    static forRoot() {
        return {
            ngModule: NgxYoutubePlayerModule_1,
            providers: [YoutubePlayerService]
        };
    }
};
NgxYoutubePlayerModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgxYoutubePlayerModule });
NgxYoutubePlayerModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgxYoutubePlayerModule_Factory(t) { return new (t || NgxYoutubePlayerModule)(); }, providers: [YoutubePlayerService], imports: [[]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxYoutubePlayerModule, { declarations: function () { return [YoutubePlayerComponent]; }, exports: function () { return [YoutubePlayerComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgxYoutubePlayerModule, [{
        type: NgModule,
        args: [{
                declarations: [YoutubePlayerComponent],
                imports: [],
                providers: [YoutubePlayerService],
                exports: [YoutubePlayerComponent]
            }]
    }], null, null); })();
export { NgxYoutubePlayerModule };
