// ng-uikit-pro-standard/esm2015/lib/pro/smoothscroll/mdb-page-scroll.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
* Created by sebastianfuss on 03.09.16.
*/
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PageScrollService } from './mdb-page-scroll.service';
import { PageScrollDirective } from './mdb-page-scroll.directive';
import * as ɵngcc0 from '@angular/core';
export class SmoothscrollModule {
    /**
     * @return {?}
     */
    static forRoot() {
        return {
            ngModule: SmoothscrollModule,
            providers: [
                { provide: PageScrollService, useClass: PageScrollService }
            ]
        };
    }
}
SmoothscrollModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: SmoothscrollModule });
SmoothscrollModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function SmoothscrollModule_Factory(t) { return new (t || SmoothscrollModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SmoothscrollModule, { declarations: function () { return [PageScrollDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [PageScrollDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SmoothscrollModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                declarations: [PageScrollDirective],
                exports: [PageScrollDirective]
            }]
    }], null, null); })();
