// ng-uikit-pro-standard/esm5/lib/free/badge/badge.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { MDBBadgeComponent } from './mdb-badge.component';
import { NgModule } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
var BadgeModule = /** @class */ (function () {
    function BadgeModule() {
    }
BadgeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: BadgeModule });
BadgeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function BadgeModule_Factory(t) { return new (t || BadgeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(BadgeModule, { declarations: function () { return [MDBBadgeComponent]; }, exports: function () { return [MDBBadgeComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(BadgeModule, [{
        type: NgModule,
        args: [{
                declarations: [MDBBadgeComponent],
                exports: [MDBBadgeComponent]
            }]
    }], function () { return []; }, null); })();
    return BadgeModule;
}());
export { BadgeModule };
