// angular-feather/esm2015/lib/feather.module.js
import { NgModule, Optional } from '@angular/core';
import { FeatherComponent } from './feather.component';
import { Icons } from './icons.provider';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './icons.provider';
export class FeatherModule {
    constructor(icons) {
        this.icons = icons;
        if (!this.icons) {
            throw new Error(`No icon provided. Make sure to use 'FeatherModule.pick({ ... })' when importing the module\n` +
                `Refer to documentation on https://github.com/michaelbazos/angular-feather`);
        }
    }
    static pick(icons) {
        return {
            ngModule: FeatherModule,
            providers: [
                { provide: Icons, multi: true, useValue: icons }
            ]
        };
    }
}
FeatherModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: FeatherModule });
FeatherModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function FeatherModule_Factory(t) { return new (t || FeatherModule)(ɵngcc0.ɵɵinject(ɵngcc1.Icons, 8)); } });
FeatherModule.ctorParameters = () => [
    { type: Icons, decorators: [{ type: Optional }] }
];
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(FeatherModule, { declarations: function () { return [FeatherComponent]; }, exports: function () { return [FeatherComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FeatherModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    FeatherComponent
                ],
                exports: [
                    FeatherComponent
                ]
            }]
    }], function () { return [{ type: ɵngcc1.Icons, decorators: [{
                type: Optional
            }] }]; }, null); })();
