// ng-uikit-pro-standard/esm5/lib/pro/tree-view/tree-view.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MdbTreeComponent } from './tree-view.component';
import { CheckboxModule } from '../../free/checkbox/checkbox.module';
import * as ɵngcc0 from '@angular/core';
var MdbTreeModule = /** @class */ (function () {
    function MdbTreeModule() {
    }
MdbTreeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: MdbTreeModule });
MdbTreeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function MdbTreeModule_Factory(t) { return new (t || MdbTreeModule)(); }, imports: [[CommonModule, CheckboxModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(MdbTreeModule, { declarations: function () { return [MdbTreeComponent]; }, imports: function () { return [CommonModule, CheckboxModule]; }, exports: function () { return [MdbTreeComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MdbTreeModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, CheckboxModule],
                declarations: [MdbTreeComponent],
                exports: [MdbTreeComponent]
            }]
    }], function () { return []; }, null); })();
    return MdbTreeModule;
}());
export { MdbTreeModule };
