// ng-uikit-pro-standard/esm2015/lib/free/waves/waves.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { WavesDirective } from './waves-effect.directive';
import * as ɵngcc0 from '@angular/core';
export class WavesModule {
    /**
     * @return {?}
     */
    static forRoot() {
        return { ngModule: WavesModule, providers: [] };
    }
}
WavesModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: WavesModule });
WavesModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function WavesModule_Factory(t) { return new (t || WavesModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(WavesModule, { declarations: function () { return [WavesDirective]; }, exports: function () { return [WavesDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(WavesModule, [{
        type: NgModule,
        args: [{
                declarations: [WavesDirective],
                exports: [WavesDirective]
            }]
    }], null, null); })();
