// ng-uikit-pro-standard/esm5/lib/pro/progressbars/progress-bars-module/index.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ProgressBarComponent } from './progressbar.component';
import * as ɵngcc0 from '@angular/core';
export { ProgressBarComponent } from './progressbar.component';
var MdProgressBarModule = /** @class */ (function () {
    function MdProgressBarModule() {
    }
    /**
     * @return {?}
     */
    MdProgressBarModule.forRoot = /**
     * @return {?}
     */
    function () {
        return {
            ngModule: MdProgressBarModule,
            providers: [],
        };
    };
MdProgressBarModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: MdProgressBarModule });
MdProgressBarModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function MdProgressBarModule_Factory(t) { return new (t || MdProgressBarModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(MdProgressBarModule, { declarations: function () { return [ProgressBarComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ProgressBarComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MdProgressBarModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                exports: [ProgressBarComponent],
                declarations: [ProgressBarComponent]
            }]
    }], function () { return []; }, null); })();
    return MdProgressBarModule;
}());
export { MdProgressBarModule };
