// ng-uikit-pro-standard/esm5/lib/pro/progressbars/index.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './progress-bars-module/index';
import * as ɵngcc2 from './progress-spinner-module/index';
import * as ɵngcc3 from './progressbar.module';
export { BarComponent } from './bar.component';
export { ProgressDirective } from './progress.directive';
export { ProgressbarComponent } from './progressbar.component';
export { ProgressbarModule } from './progressbar.module';
export { ProgressbarConfigComponent } from './progressbar.config.component';
export { ProgressSpinnerComponent } from './progress-spinner.component';
import { ProgressbarModule } from './progressbar.module';
import { MdProgressSpinnerModule } from './progress-spinner-module/index';
import { MdProgressBarModule } from './progress-bars-module/index';
/** @type {?} */
var MATERIAL_MODULES = [MdProgressBarModule, MdProgressSpinnerModule, ProgressbarModule];
var PreloadersModule = /** @class */ (function () {
    function PreloadersModule() {
    }
PreloadersModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: PreloadersModule });
PreloadersModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function PreloadersModule_Factory(t) { return new (t || PreloadersModule)(); }, imports: [[
            MdProgressBarModule.forRoot(),
            MdProgressSpinnerModule.forRoot(),
            ProgressbarModule.forRoot(),
        ], MdProgressBarModule, MdProgressSpinnerModule, ProgressbarModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(PreloadersModule, { imports: [ɵngcc1.MdProgressBarModule, ɵngcc2.MdProgressSpinnerModule, ɵngcc3.ProgressbarModule], exports: [MdProgressBarModule, MdProgressSpinnerModule, ProgressbarModule] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PreloadersModule, [{
        type: NgModule,
        args: [{
                imports: [
                    MdProgressBarModule.forRoot(),
                    MdProgressSpinnerModule.forRoot(),
                    ProgressbarModule.forRoot(),
                ],
                exports: MATERIAL_MODULES
            }]
    }], function () { return []; }, null); })();
    return PreloadersModule;
}());
export { PreloadersModule };
var ProgressBars = /** @class */ (function () {
    function ProgressBars() {
    }
    /**
     * @return {?}
     */
    ProgressBars.forRoot = /**
     * @return {?}
     */
    function () {
        return { ngModule: PreloadersModule };
    };
ProgressBars.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ProgressBars });
ProgressBars.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ProgressBars_Factory(t) { return new (t || ProgressBars)(); }, imports: [MATERIAL_MODULES, MdProgressBarModule, MdProgressSpinnerModule, ProgressbarModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ProgressBars, { imports: [MdProgressBarModule, MdProgressSpinnerModule, ProgressbarModule], exports: [MdProgressBarModule, MdProgressSpinnerModule, ProgressbarModule] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ProgressBars, [{
        type: NgModule,
        args: [{
                imports: MATERIAL_MODULES,
                exports: MATERIAL_MODULES
            }]
    }], function () { return []; }, null); })();
    return ProgressBars;
}());
export { ProgressBars };
