// ng-uikit-pro-standard/esm5/lib/pro/inputs/char-counter.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { CharCounterDirective } from './char-counter.directive';
import * as ɵngcc0 from '@angular/core';
var CharCounterModule = /** @class */ (function () {
    function CharCounterModule() {
    }
    /**
     * @return {?}
     */
    CharCounterModule.forRoot = /**
     * @return {?}
     */
    function () {
        return { ngModule: CharCounterModule, providers: [] };
    };
CharCounterModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: CharCounterModule });
CharCounterModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function CharCounterModule_Factory(t) { return new (t || CharCounterModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(CharCounterModule, { declarations: function () { return [CharCounterDirective]; }, exports: function () { return [CharCounterDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CharCounterModule, [{
        type: NgModule,
        args: [{
                declarations: [CharCounterDirective],
                exports: [CharCounterDirective]
            }]
    }], function () { return []; }, null); })();
    return CharCounterModule;
}());
export { CharCounterModule };
