// ng-uikit-pro-standard/esm5/lib/pro/inputs/char-counter.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Directive, ElementRef, Renderer2, Input, HostListener } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
var CharCounterDirective = /** @class */ (function () {
    function CharCounterDirective(_elRef, _renderer) {
        this._elRef = _elRef;
        this._renderer = _renderer;
        this.length = 20;
    }
    /**
     * @return {?}
     */
    CharCounterDirective.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        // Inititalise a new <span> element for the count display and render it below the host component.
        this.textContainer = this._renderer.createElement('p');
        this._renderer.appendChild(this._elRef.nativeElement.parentElement, this.textContainer);
        this._renderer.addClass(this.textContainer, 'chars');
        this.textContainer.innerHTML = '0/' + this.length;
        this._renderer.setStyle(this.textContainer, 'display', 'none');
    };
    /**
     * @return {?}
     */
    CharCounterDirective.prototype.onKeyUp = /**
     * @return {?}
     */
    function () {
        this.textContainer.innerHTML = this._elRef.nativeElement.value.length + '/' + this.length;
        if (this._elRef.nativeElement.value.length > this.length) {
            this._renderer.addClass(this._elRef.nativeElement, 'invalid');
        }
        else {
            this._renderer.removeClass(this._elRef.nativeElement, 'invalid');
        }
    };
    /**
     * @return {?}
     */
    CharCounterDirective.prototype.hide = /**
     * @return {?}
     */
    function () {
        this._renderer.setStyle(this.textContainer, 'display', 'none');
    };
    /**
     * @return {?}
     */
    CharCounterDirective.prototype.show = /**
     * @return {?}
     */
    function () {
        this.onKeyUp();
        this._renderer.setStyle(this.textContainer, 'display', 'block');
    };
    /** @nocollapse */
    CharCounterDirective.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer2 }
    ]; };
    CharCounterDirective.propDecorators = {
        length: [{ type: Input }],
        onKeyUp: [{ type: HostListener, args: ['input',] }],
        hide: [{ type: HostListener, args: ['blur',] }],
        show: [{ type: HostListener, args: ['focus',] }]
    };
CharCounterDirective.ɵfac = function CharCounterDirective_Factory(t) { return new (t || CharCounterDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2)); };
CharCounterDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: CharCounterDirective, selectors: [["", "mdbCharCounter", ""]], hostBindings: function CharCounterDirective_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("input", function CharCounterDirective_input_HostBindingHandler() { return ctx.onKeyUp(); })("blur", function CharCounterDirective_blur_HostBindingHandler() { return ctx.hide(); })("focus", function CharCounterDirective_focus_HostBindingHandler() { return ctx.show(); });
    } }, inputs: { length: "length" } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CharCounterDirective, [{
        type: Directive,
        args: [{
                selector: '[mdbCharCounter]'
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ɵngcc0.Renderer2 }]; }, { length: [{
            type: Input
        }], onKeyUp: [{
            type: HostListener,
            args: ['input']
        }], hide: [{
            type: HostListener,
            args: ['blur']
        }], show: [{
            type: HostListener,
            args: ['focus']
        }] }); })();
    return CharCounterDirective;
}());
export { CharCounterDirective };
if (false) {
    /** @type {?} */
    CharCounterDirective.prototype.length;
    /** @type {?} */
    CharCounterDirective.prototype.textContainer;
    /**
     * @type {?}
     * @private
     */
    CharCounterDirective.prototype._elRef;
    /**
     * @type {?}
     * @private
     */
    CharCounterDirective.prototype._renderer;
}
