// ng-uikit-pro-standard/esm5/lib/pro/range/range.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { FormsModule } from '@angular/forms';
import { CommonModule } from '@angular/common';
import { MdbRangeInputComponent } from './mdb-range.component';
import { NgModule } from '@angular/core';
import { MdbMultiRangeInputComponent } from './multi-range/mdb-multi-range.component';
import * as ɵngcc0 from '@angular/core';
var RangeModule = /** @class */ (function () {
    function RangeModule() {
    }
RangeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: RangeModule });
RangeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function RangeModule_Factory(t) { return new (t || RangeModule)(); }, imports: [[CommonModule, FormsModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(RangeModule, { declarations: function () { return [MdbRangeInputComponent, MdbMultiRangeInputComponent]; }, imports: function () { return [CommonModule, FormsModule]; }, exports: function () { return [MdbRangeInputComponent, MdbMultiRangeInputComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RangeModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, FormsModule],
                declarations: [MdbRangeInputComponent, MdbMultiRangeInputComponent],
                exports: [MdbRangeInputComponent, MdbMultiRangeInputComponent]
            }]
    }], function () { return []; }, null); })();
    return RangeModule;
}());
export { RangeModule };
