// ng-uikit-pro-standard/esm2015/lib/free/carousel/carousel.config.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class CarouselConfig {
    constructor() {
        /**
         * Default interval of auto changing of slides
         */
        this.interval = 5000;
        /**
         * Is loop of auto changing of slides can be paused
         */
        this.noPause = false;
        /**
         * Is slides can wrap from the last to the first slide
         */
        this.noWrap = false;
        this.keyboard = false;
    }
}
CarouselConfig.ɵfac = function CarouselConfig_Factory(t) { return new (t || CarouselConfig)(); };
CarouselConfig.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: CarouselConfig, factory: CarouselConfig.ɵfac });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CarouselConfig, [{
        type: Injectable
    }], function () { return []; }, null); })();
if (false) {
    /**
     * Default interval of auto changing of slides
     * @type {?}
     */
    CarouselConfig.prototype.interval;
    /**
     * Is loop of auto changing of slides can be paused
     * @type {?}
     */
    CarouselConfig.prototype.noPause;
    /**
     * Is slides can wrap from the last to the first slide
     * @type {?}
     */
    CarouselConfig.prototype.noWrap;
    /** @type {?} */
    CarouselConfig.prototype.keyboard;
}
