// ng-uikit-pro-standard/esm2015/lib/pro/easy-charts/chart-simple.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { SimpleChartComponent } from './chart-simple.component';
import { EasyPieChartComponent } from './chart-smallpie.component';
import * as ɵngcc0 from '@angular/core';
export class ChartSimpleModule {
}
ChartSimpleModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ChartSimpleModule });
ChartSimpleModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ChartSimpleModule_Factory(t) { return new (t || ChartSimpleModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ChartSimpleModule, { declarations: function () { return [SimpleChartComponent, EasyPieChartComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [SimpleChartComponent, EasyPieChartComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ChartSimpleModule, [{
        type: NgModule,
        args: [{
                declarations: [SimpleChartComponent, EasyPieChartComponent],
                exports: [SimpleChartComponent, EasyPieChartComponent],
                imports: [CommonModule]
            }]
    }], null, null); })();
