// ng-uikit-pro-standard/esm2015/lib/pro/time-picker/timepicker.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { NgModule } from '@angular/core';
import { ClockPickerComponent } from './timepicker.component';
import { ButtonsModule } from '../../free/buttons/buttons.module';
import { WavesModule } from '../../free/waves/waves.module';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../free/buttons/buttons.module';
import * as ɵngcc2 from '../../free/waves/waves.module';
export class TimePickerModule {
}
TimePickerModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: TimePickerModule });
TimePickerModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function TimePickerModule_Factory(t) { return new (t || TimePickerModule)(); }, imports: [[CommonModule, FormsModule, ButtonsModule.forRoot(), WavesModule.forRoot()]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(TimePickerModule, { declarations: function () { return [ClockPickerComponent]; }, imports: function () { return [CommonModule, FormsModule, ɵngcc1.ButtonsModule, ɵngcc2.WavesModule]; }, exports: function () { return [ClockPickerComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TimePickerModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, FormsModule, ButtonsModule.forRoot(), WavesModule.forRoot()],
                declarations: [ClockPickerComponent],
                exports: [ClockPickerComponent]
            }]
    }], null, null); })();
