// ng-uikit-pro-standard/esm5/lib/free/input-utilities/input-utilities.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MdbErrorDirective } from './error.directive';
import { MdbSuccessDirective } from './success.directive';
import { MdbValidateDirective } from './validate.directive';
import * as ɵngcc0 from '@angular/core';
var InputUtilitiesModule = /** @class */ (function () {
    function InputUtilitiesModule() {
    }
InputUtilitiesModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: InputUtilitiesModule });
InputUtilitiesModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function InputUtilitiesModule_Factory(t) { return new (t || InputUtilitiesModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(InputUtilitiesModule, { declarations: function () { return [MdbErrorDirective, MdbSuccessDirective, MdbValidateDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [MdbErrorDirective, MdbSuccessDirective, MdbValidateDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(InputUtilitiesModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                declarations: [MdbErrorDirective, MdbSuccessDirective, MdbValidateDirective],
                exports: [MdbErrorDirective, MdbSuccessDirective, MdbValidateDirective]
            }]
    }], function () { return []; }, null); })();
    return InputUtilitiesModule;
}());
export { InputUtilitiesModule };
