// ngx-typed-js/esm2015/lib/ngx-typed-js.module.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { NgxTypedJsComponent } from './ngx-typed-js.component';
import * as ɵngcc0 from '@angular/core';
let NgxTypedJsModule = class NgxTypedJsModule {
};
NgxTypedJsModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgxTypedJsModule });
NgxTypedJsModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgxTypedJsModule_Factory(t) { return new (t || NgxTypedJsModule)(); }, imports: [[]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxTypedJsModule, { declarations: function () { return [NgxTypedJsComponent]; }, exports: function () { return [NgxTypedJsComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgxTypedJsModule, [{
        type: NgModule,
        args: [{
                declarations: [NgxTypedJsComponent],
                imports: [],
                exports: [NgxTypedJsComponent]
            }]
    }], null, null); })();
export { NgxTypedJsModule };
